R/data_rotate.R

```diff
@@ -0,0 +1,63 @@
+#' @title Rotate a dataframe
+#' @name data_rotate
+#'
+#' @description
+#' This function rotates a dataframe, i.e. columns become rows and vice versa.
+#'
+#' @param data A dataframe.
+#' @param rownames Character vector (optional). If not `NULL`, the dataframe's
+#'   rownames will be added as (first) column to the output, with `rownames`
+#'   being the name of this column.
+#' @param colnames Logical (optional), if `TRUE`, the values of the first column
+#'   in `x` will be used as column names in the rotated dataframe.
+#'
+#' @return A (rotated) dataframe.
+#'
+#' @examples
+#' x <- mtcars[1:3, 1:4]
+#' data_rotate(x)
+#' data_rotate(x, rownames = "property")
+#'
+#' # use values in 1. column as column name
+#' data_rotate(x, colnames = TRUE)
+#' data_rotate(x, rownames = "property", colnames = TRUE)
+#' @export
+data_rotate <- function(data, rownames = NULL, colnames = FALSE) {
+  # check if first column has column names to be used for rotated data
+  if (colnames) {
+    colnames <- data[[1]]
+    data <- data[-1]
+  } else {
+    colnames <- NULL
+  }
+
+  # copy attributes
+  a <- attributes(data)
+
+  # rotate dataframe by 90 degrees
+  data <- as.data.frame(t(as.data.frame(data)))
+
+  # add column names, if requested
+  if (!is.null(colnames)) {
+    # check if we have correct length of column names
+    if (length(colnames) != ncol(data)) {
+      warning(
```

Comment:
@strengejacke Can you think of an example where this warning might be appropriate? 
Will be good to test this condition.

Reply:
Hm, not sure what the reason was to include this check... Maybe one column data frames?

Reply:
Hmm, even in that context no warning is issued:

``` r
datawizard::data_rotate(dplyr::select(mtcars, wt), colnames = TRUE)
#>  [1] 2.62  2.875 2.32  3.215 3.44  3.46  3.57  3.19  3.15  3.44  3.44  4.07 
#> [13] 3.73  3.78  5.25  5.424 5.345 2.2   1.615 1.835 2.465 3.52  3.435 3.84 
#> [25] 3.845 1.935 2.14  1.513 3.17  2.77  3.57  2.78 
#> <0 rows> (or 0-length row.names)
```

<sup>Created on 2022-02-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>

Reply:
yeah. Maybe it's only useful if we would allow a character vector for `colnames` as well.

Reply:
Yes, definitely useful in that context. I will implement that change in a separate PR.